src/preferences/dialog/dlgprefrecord.cpp

```diff
@@ -247,9 +355,106 @@ void DlgPrefRecord::slotBrowseRecordingsDir() {
+    // (engine) samplerate, we check the usedefault radio button. If not,
+    // we select the correct index from the combo-box. If the combo-box
+    // does not have the value either, we must conclude that this
+    // samplerate value is not common to the old and new formats.
+    // Warn that the recording samplerate will have to be
+    // changed, and allow the user to select a samplerate
+    // from the new custom list. This gui action triggers
+    // slotsampleratechanged and all is good.
+    int recSampleRateIdx = recSampleRateComboBox->findData(
+            QVariant::fromValue(m_oldRecSampleRate));
+    if (recSampleRateIdx != -1) { // found in the combobox
+        qDebug() << "Samplerate " << m_oldRecSampleRate << " found in combo-box";
+        RadioButtonUseDefaultSampleRate->setChecked(false);       // calls slot ignore
+        recSampleRateComboBox->setCurrentIndex(recSampleRateIdx); // calls slot sampleratechanged
+        RadioButtonUseCustomSampleRate->setChecked(true);
+        qDebug() << "checked custom samplerate radiobutton";
+    } else {
+        // If the updated GUI list is missing the old rec samplerate,
+        // that samplerate is the new default, i.e. engine samplerate.
+        // select the default radio button instead.
+        // However it could also be the case that even the default samplerate
+        // is not supported. Need to check the value of default samplerate.
+        qDebug() << "Default rate: " << m_defaultSampleRate << " old rate: " << m_oldRecSampleRate;
+        if (isDefaultRateValid) {
+            RadioButtonUseDefaultSampleRate->setEnabled(true);
+            if (m_defaultSampleRate == m_oldRecSampleRate) {
+                RadioButtonUseDefaultSampleRate->setChecked(true);
+            } else {
+                RadioButtonUseCustomSampleRate->setChecked(true);
+            }
+        } else {
+            RadioButtonUseDefaultSampleRate->setEnabled(false);
+            RadioButtonUseDefaultSampleRate->setChecked(false);
+            RadioButtonUseCustomSampleRate->setChecked(true);
+            qWarning() << "The current format does not support the original "
+                          "recording samplerate ("
+                       << m_oldRecSampleRate << "Hz)";
+        }
+    }
     setupEncoderUI();
 }
 
+// Update the set of custom sample rates offered in
+// the recording combo-box.
+// called on 2 occasions during a Mixxx run:
+// 1. From a slot, whenever the engine samplerate is updated.
+// 2. From a slot, whenever the recording format is changed.
+void DlgPrefRecord::updateSampleRates(
+        const QList<mixxx::audio::SampleRate>& sampleRates) {
+    recSampleRateComboBox->blockSignals(true);
+
+    recSampleRateComboBox->clear();
+    for (const auto& sampleRate : sampleRates) {
+        if (sampleRate.isValid()) {
+            if (sampleRate.value() == m_defaultSampleRate) {
+                continue; // do not add default samplerate to list.
```

Comment:
I think we have a use case for it. If a user wants to force 44100 Hz and independent from the engine sample rate.